tools/cnvkit_import_theta2.cwl

```diff
@@ -10,45 +10,52 @@ requirements:
+
+      cnvkit.py import-theta
+      ${if (inputs.tumor_cns != null) {return inputs.tumor_cns.path;}}
+      ${if (inputs.theta2_n2_results != null) {return inputs.theta2_n2_results.path;}}
       -d ./
 
-      mv $(inputs.output_basename).call-1.cns $(inputs.output_basename).theta2.total.cns
+      mv ${return inputs.output_basename}.call-1.cns ${return inputs.output_basename}.theta2.total.cns
       
-      ln -s $(inputs.output_basename).theta2.total.cns $(inputs.tumor_sample_name).cns
+      ln -s ${return inputs.output_basename}.theta2.total.cns ${return inputs.tumor_sample_name}.cns
 
-      cnvkit.py export seg $(inputs.tumor_sample_name).cns -o $(inputs.output_basename).theta2.total.seg
+      cnvkit.py export seg ${return inputs.tumor_sample_name}.cns -o ${return inputs.output_basename}.theta2.total.seg
 
-      rm $(inputs.tumor_sample_name).cns
+      rm ${return inputs.tumor_sample_name}.cns
 
       cnvkit.py import-theta
-      $(inputs.tumor_cns.path)
-      $(inputs.theta2_best_results.path)
+      ${if (inputs.tumor_cns != null) {return inputs.tumor_cns.path;}}
+      ${if (inputs.theta2_best_results != null) {return inputs.theta2_best_results.path;}}
       -d ./
 
-      mv $(inputs.output_basename).call-1.cns $(inputs.output_basename).theta2.subclone1.cns
+      mv ${return inputs.output_basename}.call-1.cns ${return inputs.output_basename}.theta2.subclone1.cns
 
-      ln -s $(inputs.output_basename).theta2.subclone1.cns $(inputs.tumor_sample_name).cns
+      ln -s ${return inputs.output_basename}.theta2.subclone1.cns ${return inputs.tumor_sample_name}.cns
 
-      cnvkit.py export seg $(inputs.tumor_sample_name).cns -o $(inputs.output_basename).theta2.subclone1.seg
+      cnvkit.py export seg ${return inputs.tumor_sample_name}.cns -o ${return inputs.output_basename}.theta2.subclone1.seg
 
-      rm $(inputs.tumor_sample_name).cns
+      rm ${return inputs.tumor_sample_name}.cns
 
-      SC2=$(inputs.output_basename).call-2.cns;
+      SC2=${return inputs.output_basename}.call-2.cns;
 
       if [ -f "$SC2" ]; then
-          mv $(inputs.output_basename).call-2.cns $(inputs.output_basename).theta2.subclone2.cns;
-          ln -s $(inputs.output_basename).theta2.subclone2.cns $(inputs.tumor_sample_name).cns;
-          cnvkit.py export seg $(inputs.tumor_sample_name).cns -o $(inputs.output_basename).theta2.subclone2.seg;
-          rm $(inputs.tumor_sample_name).cns;
+          mv ${return inputs.output_basename}.call-2.cns ${return inputs.output_basename}.theta2.subclone2.cns;
+          ln -s ${return inputs.output_basename}.theta2.subclone2.cns ${return inputs.tumor_sample_name}.cns;
+          cnvkit.py export seg ${return inputs.tumor_sample_name}.cns -o ${return inputs.output_basename}.theta2.subclone2.seg;
+          rm ${return inputs.tumor_sample_name}.cns;
       else
-        echo "second subclone file not found.  skipping!";
+        echo "second subclone file not found.  skipping! >&2;";
       fi
 
 inputs:
```

Comment:
Don't the file inputs need to be optional in order to have `null` values for:
```
inputs.theta2_n2_results
inputs.tumor_cns
inputs.theta2_best_results
```
not crash the pipeline?

Reply:
Oddly enough, it didn't, but I will add that anyway to make the pipel;ine cleaner.  Good catch!

Reply:
Pushed changes